NPE when trying to list files

## Exception
* __User Action:__ UI Error
* __Request:__ Application crash
* __Version:__ 3.5.2
* __OS:__ Linux HUAWEI/AGS2-W09EEA/HWAGS2:8.0.0/HUAWEIAGS2-W09/310(OCEC431):user/release-keys 8.0.0 - 26
* __Device:__ HWAGS2
* __Model:__ AGS2-W09
* __Product:__ AGS2-W09EEA
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
at android.os.AsyncTask$3.done(AsyncTask.java:365)
at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
at java.util.concurrent.FutureTask.run(FutureTask.java:271)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
at java.lang.Thread.run(Thread.java:784)
Caused by: java.lang.NullPointerException: pathFile.listFiles() must not be null
at com.amaze.filemanager.filesystem.root.ListFilesCommand.getFilesList(ListFilesCommand.kt:155)
at com.amaze.filemanager.filesystem.root.ListFilesCommand.listFiles(ListFilesCommand.kt:67)
at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.doInBackground(LoadFilesListTask.java:222)
at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.doInBackground(LoadFilesListTask.java:67)
at android.os.AsyncTask$2.call(AsyncTask.java:345)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
... 3 more

```
</details>
<hr>


PLEASE HELP. My computer doesn't recognise my external hard drive.

--- a/app/src/main/java/com/amaze/filemanager/filesystem/root/ListFilesCommand.kt
+++ b/app/src/main/java/com/amaze/filemanager/filesystem/root/ListFilesCommand.kt
@@ -22,6 +22,7 @@ package com.amaze.filemanager.filesystem.root
 
 import android.preference.PreferenceManager
 import android.util.Log
+import com.amaze.filemanager.R
 import com.amaze.filemanager.application.AppConfig
 import com.amaze.filemanager.exceptions.ShellCommandInvalidException
 import com.amaze.filemanager.exceptions.ShellNotRunningException
@@ -36,6 +37,8 @@ import kotlin.collections.ArrayList
 
 object ListFilesCommand : IRootCommand() {
 
+    private val TAG: String = javaClass.simpleName
+
     /**
      * list files in given directory and invoke callback
      */
@@ -46,7 +49,7 @@ object ListFilesCommand : IRootCommand() {
         openModeCallback: (openMode: OpenMode) -> Unit,
         onFileFoundCallback: (file: HybridFileParcelable) -> Unit
     ) {
-        var mode: OpenMode
+        val mode: OpenMode
         if (root && !path.startsWith("/storage") && !path.startsWith("/sdcard")) {
             // we're rooted and we're trying to load file with superuser
             // we're at the root directories, superuser is required!
@@ -145,34 +148,42 @@ object ListFilesCommand : IRootCommand() {
      * @param path the path
      */
     private fun getFilesList(
-        path: String?,
+        path: String,
         showHidden: Boolean,
         listener: (HybridFileParcelable) -> Unit
-    ): ArrayList<HybridFileParcelable>? {
+    ): ArrayList<HybridFileParcelable> {
         val pathFile = File(path)
         val files = ArrayList<HybridFileParcelable>()
         if (pathFile.exists() && pathFile.isDirectory) {
-            pathFile.listFiles().forEach { currentFile ->
-                var size: Long = 0
-                if (!currentFile.isDirectory) size = currentFile.length()
-                HybridFileParcelable(
-                    currentFile.path,
-                    RootHelper.parseFilePermission(currentFile),
-                    currentFile.lastModified(),
-                    size,
-                    currentFile.isDirectory
-                ).let { baseFile ->
-                    baseFile.name = currentFile.name
-                    baseFile.mode = OpenMode.FILE
-                    if (showHidden) {
-                        files.add(baseFile)
-                        listener(baseFile)
-                    } else {
-                        if (!currentFile.isHidden) {
+            val filesInPathFile = pathFile.listFiles()
+            if (filesInPathFile != null) {
+                filesInPathFile.forEach { currentFile ->
+                    var size: Long = 0
+                    if (!currentFile.isDirectory) size = currentFile.length()
+                    HybridFileParcelable(
+                        currentFile.path,
+                        RootHelper.parseFilePermission(currentFile),
+                        currentFile.lastModified(),
+                        size,
+                        currentFile.isDirectory
+                    ).let { baseFile ->
+                        baseFile.name = currentFile.name
+                        baseFile.mode = OpenMode.FILE
+                        if (showHidden) {
                             files.add(baseFile)
                             listener(baseFile)
+                        } else {
+                            if (!currentFile.isHidden) {
+                                files.add(baseFile)
+                                listener(baseFile)
+                            }
                         }
                     }
+                }
+            } else {
+                Log.e(TAG, "Error listing files at [$path]. Access permission denied?")
+                AppConfig.getInstance().run {
+                    AppConfig.toast(this, this.getString(R.string.error_permission_denied))
                 }
             }
         }

--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -697,5 +697,6 @@
     <string name="clear_open_file_summary">Clears selected default file opening apps</string>
     <string name="legacy_listing_title">Use legacy listing for root</string>
     <string name="legacy_listing_summary">If enabled, uses stat for listing, else uses ls</string>
+    <string name="error_permission_denied">Permission denied</string>
 </resources>
 

